### PR TITLE
 Add the support of CREATE SEQUENCE syntax

### LINKED ISSUE #60: [Feature] Add the support of CREATE SEQUENCE
Spanner introduced the DDL SEQUENCE and UUID as part of a primary key DEFAULT expression, please refer: https://cloud.google.com/spanner/docs/release-notes#August_23_2023

I would like to support this feature if no other guys have already been on it.

### PR DESCRIPTION
**Pull request by git-hulk**
This closes #60

### CODE REVIEW

File: ast/ast.go [hunk 1]
```diff
@@ -1305,6 +1307,22 @@ type CreateTable struct {
 	RowDeletionPolicy *CreateRowDeletionPolicy // optional
 }
 
+// CreateSequence is CREATE SEQUENCE statement node.
+// CREATE SEQUENCE IF NOT EXISTS {{.Name | sql}} }}
+// OPTIONS (
+//
+//	{{.Options | sqlJoin ","}}
+//
+// )
```
Comment (makenowjust) [inline ast/ast.go:1316]:
```suggestion
// CreateSequence is CREATE SEQUENCE statement node.
//
//	CREATE SEQUENCE IF NOT EXISTS {{.Name | sql}} }} OPTIONS ({{.Options | sqlJoin ","}})
```

File: ast/ast.go [hunk 2]
```diff
@@ -1305,6 +1307,22 @@ type CreateTable struct {
...
+type CreateSequence struct {
+	// pos = Create
+	Create      token.Pos // position of "CREATE" keyword
+	Rparen      token.Pos // position of ")" of OPTIONS clause
+	Name        *Ident
+	IfNotExists bool
+	Options     []*SequenceOption
```
Comment (makenowjust) [inline ast/ast.go:1323]: I'm not sure `len(Options) > 0` is actually correct. Is this right?

```suggestion
	// pos = Create
	// end = Rparen + 1

	Create      token.Pos // position of "CREATE" keyword
	Rparen      token.Pos // position of ")" of OPTIONS clause

	Name        *Ident
	IfNotExists bool
	Options     []*SequenceOption // len(Options) > 0
```

File: ast/ast.go [hunk 3]
```diff
@@ -1920,3 +1938,13 @@ type UpdateItem struct {
 	Path []*Ident // len(Path) > 0
 	Expr Expr
 }
+
+// SequenceOption is option for CREATE SEQUENCE.
+//
+// {{.Name | sql}} = {{.Value | sql}}
```
Comment (makenowjust) [inline ast/ast.go:1944]:
```suggestion
//	{{.Name | sql}} = {{.Value | sql}}
```

File: ast/ast.go [hunk 4]
```diff
@@ -1920,3 +1938,13 @@ type UpdateItem struct {
...
+type SequenceOption struct {
+	// pos = Name.pos
+	// end = Value.pos
+	Name  *Ident
+	Value Expr
```
Comment (makenowjust) [inline ast/ast.go:1949]:
```suggestion
	// pos = Name.pos
	// end = Value.end

	Name  *Ident
	Value Expr
```

**Review by makenowjust — CHANGES_REQUESTED** [review]

File: ast/ast.go (hunk 2 above)
Comment (git-hulk) [reply ast/ast.go:1319]: Yes, you're right. len(Options) should be always > 0.

Comment (makenowjust) [reply ast/ast.go]: Thank you. Such comments on invariants are useful for code reading.

**Review by makenowjust — COMMENTED** [review]
Updating `ast_test.go` is missing?

**Comment by git-hulk** [comment]
> Updating `ast_test.go` is missing?

Yes, good catch. It's done now.

**Review by makenowjust — APPROVED** [review]
Perfect. Thank you.

**Pull request merged**